compile.sh

```diff
@@ -1,41 +1,151 @@
 #!/bin/sh
+set -e
```

Comment:
If any of the commands return an error, the script will terminate instead of just continuing which is the default bash behaviour.

Why did you move it though?